kernel/src/sched/priority.rs

```diff
@@ -0,0 +1,122 @@
+// SPDX-License-Identifier: MPL-2.0
+
+use core::sync::atomic::{AtomicI8, AtomicU8};
+
+/// The process scheduling nice value.
+///
+/// It is an integer in the range of [-20, 19]. Process with a smaller nice
+/// value is more favorable in scheduling.
+pub type Nice = RangedI8<-20, 19>;
+
+impl Default for Nice {
+    fn default() -> Self {
+        Self::new(0)
+    }
+}
+
+/// The thread scheduling priority value.
+///
+/// It is an integer in the range of [0, 139]. Thread with a higher priority
+/// value is more favorable in scheduling.
+pub type Priority = RangedU8<0, 139>;
+
+impl From<Nice> for Priority {
+    fn from(value: Nice) -> Self {
```

Comment:
Thank you for pointing this out. The latest version follows what you proposed with an exception that I didn't implement the `unwrap` method because I don't get what it's trying to do. A typical user of `Priority` would expect she can create, get and set it, for which the first three methods are just enough.